pkg/testexecutionservice/testexecution.go

```diff
@@ -308,12 +309,22 @@ func (tes *testExecutionService) buildCmdArgsV1(ctx context.Context,
 	tasConfig *core.TASConfig,
 	payload *core.Payload,
 	target []string) ([]string, error) {
-	args := []string{global.FrameworkRunnerMap[tasConfig.Framework], "--command", "execute"}
+	args := []string{global.FrameworkRunnerMap[tasConfig.Framework]}
+	language := global.FrameworkLanguageMap[tasConfig.Framework]
+
+	if language == "java" {
+		args = append(args, "-jar", "/test-at-scale-java.jar",
+			global.LangArgKeyMap["command"], "execute", global.LangArgKeyMap["frameworkVersion"],
+			strconv.Itoa(tasConfig.FrameworkVersion))
+	} else {
+		args = append(args, global.LangArgKeyMap["command"], "execute")
+	}
```

Comment:
duplicate lines in discovery and execution. Can we refactor them?

Reply:
separated out the common part in a new function.